packages/connect-session/src/utils/verify-decode-id-token.ts

```diff
@@ -47,55 +20,31 @@ interface Claim {
   client_id: string
 }
 
-let cacheKeys: MapOfKidToPublicKey | undefined
-
-const getPublicKeys = async (connectUserPoolId: string): Promise<MapOfKidToPublicKey | undefined> => {
-  if (cacheKeys) return cacheKeys
-
-  try {
-    const cognitoIssuer = `https://cognito-idp.eu-west-2.amazonaws.com/${connectUserPoolId}/.well-known/jwks.json`
-    const res = await fetch(cognitoIssuer, {
-      method: 'GET',
-      headers: { 'Content-Type': 'application/json' },
-    })
-
-    const publicKeys: PublicKeys = await res.json()
-
-    if (!publicKeys) throw new Error('Public keys not returned from Reapit Connect')
-
-    cacheKeys = publicKeys.keys.reduce((agg, current) => {
-      const pem = jwkToPem(current as RSA)
-
-      agg[current.kid] = { instance: current, pem }
-
-      return agg
-    }, {} as MapOfKidToPublicKey)
-    return cacheKeys as MapOfKidToPublicKey
-  } catch (error) {
-    console.error('Reapit Connect Session error:', error.message)
-  }
-}
-
 export const connectSessionVerifyDecodeIdTokenWithPublicKeys = async (
   token: string,
   connectUserPoolId: string,
-  keys: MapOfKidToPublicKey,
 ): Promise<LoginIdentity | undefined> => {
   try {
-    const tokenSections = token.split('.')
     const cognitoIssuer = `https://cognito-idp.eu-west-2.amazonaws.com/${connectUserPoolId}`
 
-    if (tokenSections.length < 2) throw new Error('Id token is invalid')
-
-    const headerJSON = Buffer.from(tokenSections[0], 'base64').toString('utf8')
+    const decodedToken = decodeJWT(token)
 
-    const header = JSON.parse(headerJSON) as TokenHeader
-
-    const key = keys[header.kid]
-
-    if (!key) throw new Error('Id verification claim made for unknown kid')
+    const verifier = new IdTokenVerifier({
+      issuer: cognitoIssuer,
```

Comment:
needs a `leeway` key: value - see https://www.npmjs.com/package/idtoken-verifier Should be set to the maximum of 5mins to handle non-unix stupid clocks